wall_e/extensions/sfu.py

```diff
@@ -502,6 +506,111 @@ async def outline(self, ctx, *course):
         if e_obj is not False:
             await ctx.send(embed=e_obj, reference=ctx.message)
 
+    @app_commands.command(name="courses", description="Gets all offered courses")
+    @app_commands.describe(department="Specify the department to search. Examples: STAT, PHYS, MATH")
+    @app_commands.describe(level="Specifies the level of courses to filter for. Examples: 100, 200, 400")
+    @app_commands.describe(term="Specify the specific semester to search for. Examples: Spring, Summer, Fall")
+    @app_commands.describe(year="Specify the specific year to search for. Examples: 2020, 2024, 2025")
+    async def courses(self, interaction: discord.Interaction, department: str = "", level: int = 0,
+                  term: str = "registration", year: str = "registration"):
+        self.logger.info(
+            f'[SFU courses()] courses command detected from user {interaction.user} with arguments: '
+            f'department {department}, level {level}, term {term}, year {year}'
+        )
+        try:
+            await interaction.response.defer()
+        except discord.errors.NotFound:
+            await interaction.channel.send(
+                "Feeling a bit overloaded at the moment...Please try again in a few minutes"
+            )
+            return
+
+        # The default course selection if not specified
+        departments = ['CMPT', 'MATH', 'MACM']
+        if department:
```

Comment:
how do you know the department the user specified is valid?

Reply:
i dont, if the user specified department does not exist or is invalid, then sfus api should not return any results and therefore print an error message

Reply:
do you know what the error message is?